bee-tangle/src/urts.rs

```diff
@@ -175,23 +176,29 @@ impl UrtsTipPool {
         Score::NonLazy
     }
 
-    pub fn two_non_lazy_tips(&self) -> Option<(MessageId, MessageId)> {
+    pub fn two_non_lazy_tips(&self) -> Option<Vec<MessageId>> {
         let non_lazy_tips = &self.non_lazy_tips;
         if non_lazy_tips.is_empty() {
             None
-        } else if non_lazy_tips.len() == 1 {
-            let tip = non_lazy_tips.iter().next().unwrap();
-            Some((*tip, *tip))
-        } else if non_lazy_tips.len() == 2 {
-            let mut iter = non_lazy_tips.iter();
-            Some((*iter.next().unwrap(), *iter.next().unwrap()))
         } else {
-            let hashes = non_lazy_tips.iter().choose_multiple(&mut rand::thread_rng(), 2);
-            let mut iter = hashes.iter();
-            Some((**iter.next().unwrap(), **iter.next().unwrap()))
+            Some(if non_lazy_tips.len() < self.optimal_num_tips() {
+                non_lazy_tips.iter().map(|t| *t).collect()
+            } else {
+                non_lazy_tips
+                    .iter()
+                    .choose_multiple(&mut rand::thread_rng(), self.optimal_num_tips())
+                    .iter()
+                    .map(|t| **t)
+                    .collect()
+            })
```

Comment:
I think this would be easier to read:
```rust
 match self.non_lazy_tips.len() {
            0 => None,
            n if n <= self.optimal_num_tips() => {
                Some(self.non_lazy_tips.iter().copied().collect())
            }, // no point in using rand if equal
            _ => {
                Some(self.non_lazy_tips
                    .iter()
                    .choose_multiple(&mut rand::thread_rng(), self.optimal_num_tips())
                    .iter()
                    .copied()
                    .copied()
                    .collect()
                    )
            }
        }
```